### DISCUSSION TITLE
Mapsui Uno Getting Started -- Install Problem

### DISCUSSION THREAD

**Discussion by CaymanIslands** [unoplatform/uno#15652] [Q&A]
I am trying to add Mapsui to the default Uno Platform App project, per the instructions at this link:
https://mapsui.com/documentation/getting-started-uno-winui.html

```
With the Startup project set to UnoApp2.Windows, I get the following build errors.
Severity	Code	Description	Project	File	Line	Suppression State
Error	WMC9999	Type universe cannot resolve assembly: Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null.	UnoApp2	C:\source\UnoApp2\UnoApp2\Presentation\MainPage.xaml	1	
Error	WMC9999	Type universe cannot resolve assembly: Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null.	UnoApp2	C:\source\UnoApp2\UnoApp2\Presentation\MainPage.xaml	1	
Error	MSB3030	Could not copy the file "C:\source\UnoApp2\UnoApp2\obj\Debug\net8.0-windows10.0.19041\Presentation\SecondPage.xaml" because it was not found.	UnoApp2.Windows	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	5254	
Error	MSB3030	Could not copy the file "C:\source\UnoApp2\UnoApp2\obj\Debug\net8.0-windows10.0.19041\Presentation\Shell.xaml" because it was not found.	UnoApp2.Windows	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	5254	
Error	MSB3030	Could not copy the file "C:\source\UnoApp2\UnoApp2\obj\Debug\net8.0-windows10.0.19041\Presentation\MainPage.xaml" because it was not found.	UnoApp2.Windows	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	5254	
```

Is this Mapsui MapControl old / incompatible with the default project of the latest version of Uno Platform? Should I be using another version of this Mapsui MapControl? 

Any advice is much appreciated.

**Comment by jeromelaban** [comment]
Thanks for the report.  Could you provide the [binlogs for this build](https://platform.uno/docs/articles/uno-builds-troubleshooting.html)?

**Reply by CaymanIslands** [reply]
Please find attached the build binlogs that VS2022 generated from a full solution build. I don't see any log output for the UnoApp2.Mobile project in there.
[binlogs.zip](https://github.com/unoplatform/uno/files/14454571/binlogs.zip)

**Reply by CaymanIslands** [reply]
Just making sure you saw this.
Is there anything else that I can provide in order to help identify the problem?

**Reply by jeromelaban** [reply]
Thanks.

By the looks of the logs, you seem to have added an unconditional reference to `SkiaSharp.Views.Uno.WinUI`, either directly or indirectly to the `net8.0-windows10` target framework.

Your binlogs don't doncain the csproj, so if you can make a binlog using the command line with `dotnet build /bl` it will help troubleshoot further.

**Reply by CaymanIslands** [reply]
Hi Jerome.

Thanks for your message.

Ok, I recreated the project, as simply as possible, as UnoApp3. I did only the following steps.

(1) I created the default UnoPlatform App in VS2022, with startup type = Recommended.

(2) I built the default app in VS2022. No errors.

(3) I followed the instructions at the link here, in order to add the MapControl to the Uno Platform default app... 
https://mapsui.com/documentation/getting-started-uno-winui.html 
Important: I only installed their package (PM> Install-Package Mapsui.Uno.WinUI -pre) one time, while the default project (in the Package Manager window) was set to "Source\UnoApp3" <-- the cross-platform code, not any specific platform.

(4) Errors ensured in the .Wasm project, so I simply removed that project, because unloading the project was not enough to stop errors in the .Wasm project. (I'll worry about those errors later.)

(5) Errors ensured in MainPage.xaml. These are the errors that I am trying to resolve now. I attempted to correct them as follows...

(A) Their namespace reference seems to be incorrect. I had to deviate from their instructions, to change their namespace reference, from this...
xmlns:mapsui="clr-namespace:Mapsui.UI.WinUI;assembly=Mapsui.UI.Uno.WinUI" 
to this..
xmlns:mapsui="using:Mapsui.UI.WinUI"

(B) Their instantiation of the MapControl uses attributes that cause errors. I had to change their line that adds the MapControl, from this...
<mapsui:MapControl x:Name="MyMap" VerticalAlignment="Stretch" HorizontalAlignment="Stretch" />
to this...
<mapsui:MapControl x:Name="MyMap" />

Now I can build (command line "dotnet build /bl") with only one error, which I don't .

Please find that .binlog file attached.
[msbuild.binlog.zip](https://github.com/unoplatform/uno/files/14502660/msbuild.binlog.zip)

And in VS2022, the errors as follows....
Severity	Code	Description	Project	File	Line	Suppression State
Error	WMC1006	Cannot resolve Assembly or Windows Metadata file 'Type universe cannot resolve assembly: Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null.'	UnoApp3	C:\source\UnoApp3\UnoApp3\UnoApp3.csproj	1	
Error	MSB3030	Could not copy the file "C:\source\UnoApp3\UnoApp3\obj\Debug\net8.0-windows10.0.19041\Presentation\SecondPage.xaml" because it was not found.	UnoApp3	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	5254	
Error	MSB3030	Could not copy the file "C:\source\UnoApp3\UnoApp3\obj\Debug\net8.0-windows10.0.19041\Presentation\MainPage.xaml" because it was not found.	UnoApp3	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	5254	
Error	MSB3030	Could not copy the file "C:\source\UnoApp3\UnoApp3\obj\Debug\net8.0-windows10.0.19041\Presentation\Shell.xaml" because it was not found.	UnoApp3	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	5254

**Reply by jeromelaban** [reply]
Thanks. Because of the way the build is setup, `dotnet build /bl` cannot work (it's a winappsdk issue), and you'll need to use `msbuild /r /bl` in a `VS developer command prompt` on windows.

Still, this is a mapsui issue (https://github.com/Mapsui/Mapsui/issues/2523), and you'll find a workaround there until a new mapsui version is released.

**Reply by CaymanIslands** [reply]
Ok, here's that binlog.
[msbuild.zip](https://github.com/unoplatform/uno/files/14531323/msbuild.zip)

**Reply by CaymanIslands** [reply]
Ok, thanks.
I pasted the workaround into UnoApp3.csproj (the common project), and then a build raises the following error...

Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1010	The PackageReference items Mapsui do not have corresponding PackageVersion.	UnoApp3	C:\source\UnoApp3\UnoApp3\UnoApp3.csproj	1	

Here is the binlog for that.
[msbuild.zip](https://github.com/unoplatform/uno/files/14594471/msbuild.zip)

It seems like it wants a <PackageVersion> attribute for Mapsui included somewhere within the workaround, but I don't know how to include this. 

Then I tried these modifications to the workaround...
[edited code.txt](https://github.com/unoplatform/uno/files/14594814/edited.code.txt)

That works, but now my namespaces aren't working in Mapsui, because they are still pointing (depending on) .net 8.0 and for example, the Mapsui.Tiling namespace is different between the two. So the workaround isn't getting me there.

This is frustrating. I guess Uno and Mapsui isn't the way to go. Thanks for looking at this though.

**Reply by jeromelaban** [reply]
A PR was added recently to fix this issue: https://github.com/Mapsui/Mapsui/pull/2535, so I guess this will be available soon.

**Comment by pauldendulk** [comment]
This should be fixed in Mapsui 4.1.5.

**Reply by jeromelaban** [reply]
Thanks Paul!
